src/test/rgw/rgw_multi/tests.py

```diff
@@ -1339,6 +1339,38 @@ def test_bucket_reshard_incremental():
         k.set_contents_from_string('foo')
     zonegroup_bucket_checkpoint(zonegroup_conns, bucket.name)
 
+@attr('bucket_reshard')
+def test_bucket_reshard_during_incremental():
+    zonegroup = realm.master_zonegroup()
+    zonegroup_conns = ZonegroupConns(zonegroup)
+    zone = zonegroup_conns.rw_zones[0]
+
+    # create a bucket in first zone
+    bucket = zone.create_bucket(gen_bucket_name())
+    log.debug('created bucket=%s', bucket.name)
+    zonegroup_meta_checkpoint(zonegroup)
+
+    # upload some objects just to get every zone into incremental
+    for objname in ('a', 'b', 'c', 'd'):
+        k = new_key(zone, bucket.name, objname)
+        k.set_contents_from_string('foo')
+    zonegroup_bucket_checkpoint(zonegroup_conns, bucket.name)
+
+    # upload more objects to other zones
+    for z in zonegroup_conns.rw_zones[1:]:
```

Comment:
in my test this is happening even if all objects are uploaded to the master zone

Reply:
ok, that's really strange. i woudn't expect the master zone to do any actual syncing in that case